custom/icds/translations/integrations/transifex.py

```diff
@@ -39,15 +39,16 @@ def __init__(self, domain, app_id, source_lang, project_slug, version=None, lang
         self.source_lang = source_lang
         self.lock_translations = lock_translations
         self.use_version_postfix = use_version_postfix
-        self.po_file_generator = POFileGenerator(domain, app_id, version, self.key_lang, source_lang, lang_prefix,
-                                                 exclude_if_default, use_version_postfix)
+        self.transifex_po_file_generator = TransifexPOFileGenerator(domain, app_id, version, self.key_lang,
+                                                                    source_lang, lang_prefix, exclude_if_default,
+                                                                    use_version_postfix)
 
     def send_translation_files(self):
         """
         submit files to transifex for performing translations
         """
         try:
-            self.po_file_generator.generate_translation_files()
+            self.transifex_po_file_generator.generate_translation_files()
             file_uploads = self._send_files_to_transifex()
             self._cleanup()
```

Comment:
Looks like you're reinventing `try`/`finally` here. This code can be simplified:
```py
pogen = self.transifex_po_file_generator
try:
    pogen.generate_translation_files()
    file_uploads = self._send_files_to_transifex()
finally:
    pogen.po_file_generator.cleanup()
```
And then you can remove the `_cleanup` method.

Also it's probably a good idea to add a `cleanup` method on `TransifexPOFileGenerator` that calls `self.po_file_generator.cleanup()` so you don't have to reach into its internal state to do cleanup here.